src/libs/api/widgetSetting.ts

```diff
@@ -92,3 +102,145 @@ export async function getWidgetSetting({
+  strategy: 'auto' | 'manual',
+  useCachedSession: boolean,
+  params: Omit<Params, 'createChannel' | 'createUserAndChannel'>
+) => {
+  type Callbacks = {
+    BotStyle: (botStyle: Response['botStyle']) => void;
+    AutoNonCached: (response: {
+      user: ResponseUser;
+      channel: ResponseChannel;
+    }) => void;
+    AutoCached: (response: { channel?: ResponseChannel }) => void;
+    ManualNonCached: (response?: { channel: ResponseChannel }) => void;
+    ManualCached: () => void;
+  };
+
+  const callbacks: {
+    onGetBotStyle: Callbacks['BotStyle'];
+    onAutoNonCached: Callbacks['AutoNonCached'];
+    onAutoCached: Callbacks['AutoCached'];
+    onManualNonCached: Callbacks['ManualNonCached'];
+    onManualCached: Callbacks['ManualCached'];
+  } = {
+    onGetBotStyle: () => {
+      /* empty */
+    },
+    onAutoNonCached: () => {
+      /* empty */
+    },
+    onAutoCached: () => {
+      /* empty */
+    },
+    onManualNonCached: () => {
+      /* empty */
+    },
+    onManualCached: () => {
+      /* empty */
+    },
+  };
+
+  const handlers = {
+    onGetBotStyle: (callback: Callbacks['BotStyle']) => {
+      callbacks.onGetBotStyle = callback;
+      return handlers;
+    },
+    onAutoNonCached: (callback: Callbacks['AutoNonCached']) => {
+      callbacks.onAutoNonCached = callback;
+      return handlers;
+    },
+    onAutoCached: (callback: Callbacks['AutoCached']) => {
+      callbacks.onAutoCached = callback;
+      return handlers;
+    },
+    onManualCached: (callback: Callbacks['ManualCached']) => {
+      callbacks.onManualCached = callback;
+      return handlers;
+    },
+    onManualNonCached: (callback: Callbacks['ManualNonCached']) => {
+      callbacks.onManualNonCached = callback;
+      return handlers;
+    },
+    get: async () => {
+      const response = await getWidgetSetting({
+        host: params.host,
+        appId: params.appId,
+        botId: params.botId,
+        ...getParamsByStrategy(strategy, useCachedSession, params),
+      });
+      callbacks.onGetBotStyle(response.botStyle);
+
+      if (strategy === 'auto') handleAutoStrategy(response);
+      if (strategy === 'manual') handleManualStrategy(response);
+    },
+  };
+
+  function handleAutoStrategy(response: Response) {
+    if (useCachedSession) {
+      callbacks.onAutoCached({ channel: response.channel });
+    } else {
+      if (response.channel && response.user) {
+        callbacks.onAutoNonCached({
+          channel: response.channel,
+          user: response.user,
+        });
+      }
+    }
+  }
+
+  function handleManualStrategy(response: Response) {
+    if (useCachedSession) {
+      callbacks.onManualCached();
+    } else {
+      if (response.channel) {
+        callbacks.onManualNonCached({ channel: response.channel });
+      } else {
+        // Legacy manual
+        callbacks.onManualNonCached();
+      }
+    }
+  }
+
+  return handlers;
+};
+
+function getParamsByStrategy(
```

Comment:
minor: What do you think about adding comment to the function?
```
/**
 * When manual strategy, regardless of cached session, userId must be injected and channel will be
 * created "manually" through sdk once sdk is ready. See initManualSession() for reference.
 */
```

Reply:
The logic for creating channels in the SDK will likely be removed. In the future, the logic for managing sessions will operate based on the shared session flow chart. Since this includes internal policies, it might be better to share it through internal documents.

And the mentioned content is noted here: https://github.com/sendbird/chat-ai-widget/pull/261/files#diff-362487901b9c6302b2d7c52add72f6440f65334a5261fd09d24ce36fb4a7970dR162-R165

Reply:
ok good!